coalib/settings/Section.py

```diff
@@ -43,6 +44,40 @@ def append_to_sections(sections,
         key, str(value), origin, from_cli=from_cli, to_append=to_append))
 
 
+def extract_aspects_from_section(section):
+    """
+    Extracts aspects and their related settings from a section and create an
+    AspectList from it.
+
+    :param section: Section object.
+    :return:        AspectList containing aspectclass instance with
+                    user-defined tastes.
+    """
+    aspects = section.get('aspects')
+    language = section.get('language')
+
+    # Skip aspects initialization if not configured in section
+    if not len(aspects):
+        return None
+
+    if not len(language):
+        raise AttributeError('Language was not found in configuration file. '
+                             'Usage of aspect-based configuration must '
+                             'include language information.')
+
+    aspect_instances = AspectList(exclude=section.get('excludes'))
+
+    for aspect in AspectList(aspects):
+        tastes_name = [k for k, _ in section.contents.items() if
+                       aspect.__name__.lower() in k.lower()]
+        tastes_to_init = {}
+        for name in tastes_name:
+            tastes_to_init[name.split('.')[-1]] = section[name]
+
+        aspect_instances.append(aspect(language, **tastes_to_init))
+    return aspect_instances
```

Comment:
@adhikasp Your decision ;) Only keep in mind that for proper use we need to override `AspectList.append` and all other manipulation methods to check for correct aspect input